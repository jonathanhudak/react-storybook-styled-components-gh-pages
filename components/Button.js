import styled from 'styled-components';

var PRIMARY_COLOR = 'limegreen';
var Button = styled.button.withConfig({
  displayName: "Button",
  componentId: "sc-13qgl51-0"
})(["background:", ";color:", ";box-sizing:border-box;font-size:1em;margin:1em;padding:0.25em 1em;border:2px solid ", ";border-radius:3px;"], function (props) {
  return props.primary ? PRIMARY_COLOR : 'white';
}, function (props) {
  return props.primary ? 'white' : PRIMARY_COLOR;
}, PRIMARY_COLOR);

export { Button };
