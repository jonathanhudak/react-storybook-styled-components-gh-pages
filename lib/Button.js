"use strict";function _interopDefault(e){return e&&"object"==typeof e&&"default"in e?e.default:e}Object.defineProperty(exports,"__esModule",{value:!0});var styled=_interopDefault(require("styled-components"));function _taggedTemplateLiteral(e,t){return t||(t=e.slice(0)),Object.freeze(Object.defineProperties(e,{raw:{value:Object.freeze(t)}}))}function _templateObject(){var e=_taggedTemplateLiteral(["\n  background: ",";\n  color: ",";\n  box-sizing: border-box;\n  font-size: 1em;\n  margin: 1em;\n  padding: 0.25em 1em;\n  border: 2px solid ",";\n  border-radius: 3px;\n"]);return _templateObject=function(){return e},e}var PRIMARY_COLOR="limegreen",Button=styled.button(_templateObject(),function(e){return e.primary?PRIMARY_COLOR:"white"},function(e){return e.primary?"white":PRIMARY_COLOR},PRIMARY_COLOR);exports.Button=Button;
