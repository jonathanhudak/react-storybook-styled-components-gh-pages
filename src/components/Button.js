import styled from 'styled-components';

const PRIMARY_COLOR = 'limegreen';

export const Button = styled.button`
  background: ${props => (props.primary ? PRIMARY_COLOR : 'white')};
  color: ${props => (props.primary ? 'white' : PRIMARY_COLOR)};
  box-sizing: border-box;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid ${PRIMARY_COLOR};
  border-radius: 3px;
`;
