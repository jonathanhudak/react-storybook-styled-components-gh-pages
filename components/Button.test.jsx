import React from 'react'
import renderer from 'react-test-renderer';
import Button from './Button';

test('Button default', () => {
  expect(renderer.create(<Button>Perform action</Button>).toJSON())
  .toMatchSnapshot();
});

test('Button primary', () => {
  expect(renderer.create(<Button primary>Perform action</Button>).toJSON())
  .toMatchSnapshot();
});