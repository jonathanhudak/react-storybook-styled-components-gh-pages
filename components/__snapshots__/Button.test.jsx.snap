// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button default 1`] = `
.c0 {
  background: white;
  color: limegreen;
  box-sizing: border-box;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
}

<button
  className="c0"
>
  Perform action
</button>
`;

exports[`Button primary 1`] = `
.c0 {
  background: limegreen;
  color: white;
  box-sizing: border-box;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
}

<button
  className="c0"
>
  Perform action
</button>
`;
